pkg/machines/components/diskAdd.jsx

```diff
@@ -0,0 +1,445 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+
+import DialogPattern from 'cockpit-components-dialog.jsx';
+import Select from "cockpit-components-select.jsx";
+
+import { mouseClick, units, convertToUnit, digitFilter, toFixedPrecision } from '../helpers.es6';
+import { volumeCreateAndAttach, attachDisk, getVm, getStoragePools } from '../actions.es6';
+
+import './diskAdd.css';
+
+const _ = cockpit.gettext;
+
+const CREATE_NEW = 'create-new';
+const USE_EXISTING = 'use-existing';
+
+function getAvailableTargets(vm) {
+    const existingTargets = Object.getOwnPropertyNames(vm.disks);
+    const targets = [];
+    let i = 0;
+    while (i < 26 && targets.length < 5) {
+        const target = `vd${String.fromCharCode(97 + i)}`;
+        if (!existingTargets.includes(target)) {
```

Comment:
So this will only work for up to 26 disks, right? After that, Linux usually keeps counting on like "sdaa", "sdab", etc. Not a blocker for merging, I'm just curious if this is already bound by something else.

Reply:
Right, it's a limit here.
From Cockpit perspective, the dialog will be empty offering no value to select.
I believe that can be easily extended if needed.